File: Exemplos/App_eXperience_Java/app/src/main/java/com/elgin/elginexperience/SitefReturn.java
package com.elgin.elginexperience;

public class SitefReturn {
    private String cODAUTORIZACAO;
    private String vIAESTABELECIMENTO;
    private String cOMPDADOSCONF;
    private String bANDEIRA;
    private String nUMPARC;
    private String cODTRANS;
    private String rEDEAUT;
    private String nSUSITEF;
    private String vIACLIENTE;
    private String vLTROCO;
    private String tIPOPARC;
    private String cODRESP;
    private String nSUHOST;

    public String vIAESTABELECIMENTO () { return this.vIAESTABELECIMENTO; }
    public void setCODTRANS (String CODTRANS){ this.cODTRANS = CODTRANS; }
    public String cODAUTORIZACAO (){ return this.cODAUTORIZACAO; }
    public String cOMPDADOSCONF (){return this.cOMPDADOSCONF; }
    public String vIACLIENTE () { return this.vIACLIENTE;}
    public String bANDEIRA(){ return this.bANDEIRA; }
    public String cODTRANS(){ return this.cODTRANS; }
    public String rEDEAUT (){ return this.rEDEAUT; }
    public String nSUSITEF(){ return this.nSUSITEF; }
    public String vLTROCO (){ return this.vLTROCO; }
    public String tIPOPARC(){ return this.tIPOPARC; }
    public String cODRESP (){ return this.cODRESP; }
    public String nSUHOST (){ return this.nSUHOST; }
    public String nUMPARC (){
        if (this.nUMPARC != null) return this.nUMPARC;
        return "";
    }
    public String NAMETRANSCODE (){
        String retorno = "Valor invalido";
        switch (this.tIPOPARC) {
            case "00":
                retorno = "A vista";
                break;
            case "01":
                retorno = "Pré-Datado";
                break;
            case "02":
                retorno = "Parcelado Loja";
                break;
            case "03":
                retorno = "Parcelado Adm";
                break;
        }
        return retorno;
    }
}

File: M8/Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/SitefReturn.kt
package com.example.e1_elgin_kotlin

class SitefReturn {
    private val cODAUTORIZACAO: String? = null
    private val vIAESTABELECIMENTO: String? = null
    private val cOMPDADOSCONF: String? = null
    private val bANDEIRA: String? = null
    private val nUMPARC: String? = null
    private var cODTRANS: String? = null
    private val rEDEAUT: String? = null
    private val nSUSITEF: String? = null
    private val vIACLIENTE: String? = null
    private val vLTROCO: String? = null
    private val tIPOPARC: String? = null
    private val cODRESP: String? = null
    private val nSUHOST: String? = null

    fun vIAESTABELECIMENTO(): String? {
        return vIAESTABELECIMENTO
    }

    fun setCODTRANS(CODTRANS: String?) {
        cODTRANS = CODTRANS
    }

    fun cODAUTORIZACAO(): String? {
        return cODAUTORIZACAO
    }

    fun cOMPDADOSCONF(): String? {
        return cOMPDADOSCONF
    }

    fun vIACLIENTE(): String? {
        return vIACLIENTE
    }

    fun bANDEIRA(): String? {
        return bANDEIRA
    }

    fun cODTRANS(): String? {
        return cODTRANS
    }

    fun rEDEAUT(): String? {
        return rEDEAUT
    }

    fun nSUSITEF(): String? {
        return nSUSITEF
    }

    fun vLTROCO(): String? {
        return vLTROCO
    }

    fun tIPOPARC(): String? {
        return tIPOPARC
    }

    fun cODRESP(): String? {
        return cODRESP
    }

    fun nSUHOST(): String? {
        return nSUHOST
    }

    fun nUMPARC(): String? {
        return nUMPARC ?: ""
    }

    fun NAMETRANSCODE(): String? {
        var retorno = "Valor invalido"
        when (tIPOPARC) {
            "00" -> retorno = "A vista"
            "01" -> retorno = "Pré-Datado"
            "02" -> retorno = "Parcelado Loja"
            "03" -> retorno = "Parcelado Adm"
        }
        return retorno
    }
}